qiling/os/linux/thread.py

```diff
@@ -360,9 +354,14 @@ def is_running(self):
     def is_blocking(self):
         return self.status == THREAD_STATUS_BLOCKING
 
-    def update_global_thread_id(self):
-        QlLinuxThread.LINUX_THREAD_ID = os.getpid()
+    def new_thread_id(self):
+        global LINUX_THREAD_ID
+        self._thread_id = LINUX_THREAD_ID
+        LINUX_THREAD_ID += 1
 
+    def update_global_thread_id(self):
+        global LINUX_THREAD_ID
+        LINUX_THREAD_ID = os.getpid()
```

Comment:
hi, i think its best we keep the pid value to qiling space. getting the value from os might be a bad idea. (i know its our own code). maybe we should start from fresh.

Reply:
Yeah that's right, I also notice it, but it's not easy to fix it so let's leave it to another PR 😀